src/testers.jl

```diff
@@ -112,61 +112,55 @@ at input point `z` to confirm that there are correct `frule` and `rrule`s provid
 
 `fkwargs` are passed to `f` as keyword arguments.
 All keyword arguments except for `fdm` and `fkwargs` are passed to `isapprox`.
+
+To use this tester for a scalar type `MyNumber <: AbstractNumber`,
+`FiniteDifferences.to_vec(::MyNumber)` must be implemented.
 """
 function test_scalar(f, z; rtol=1e-9, atol=1e-9, fdm=_fdm, fkwargs=NamedTuple(), kwargs...)
     _ensure_not_running_on_functor(f, "test_scalar")
-    # z = x + im * y
-    # Ω = u(x, y) + im * v(x, y)
     Ω = f(z; fkwargs...)
 
+    vz, z_from_vec = to_vec(z)
+    # orthonormal tangent vectors
+    vz_basis = Diagonal(ones(eltype(vz), length(vz)))
+    Δzs = [z_from_vec(vz_basis[:, i]) for i in axes(vz_basis, 2)]
+
     # test jacobian using forward mode
-    Δx = one(z)
-    @testset "$f at $z, with tangent $Δx" begin
-        # check ∂u_∂x and (if Ω is complex) ∂v_∂x via forward mode
-        frule_test(f, (z, Δx); rtol=rtol, atol=atol, fdm=fdm, fkwargs=fkwargs, kwargs...)
-        if z isa Complex
-            # check that same tangent is produced for tangent 1.0 and 1.0 + 0.0im
+    @testset "$f at $z, with tangent $Δz" for (i, Δz) in enumerate(Δzs)
+        frule_test(f, (z, Δz); rtol=rtol, atol=atol, fdm=fdm, fkwargs=fkwargs, kwargs...)
+        if !isa(Δz, Real) && i == 1
```

Comment:
Shouldn't this be:
```suggestion
        if !isa(Δz, Real) && length(Δzs) == 1
```

Reply:
In this case, no. `i == 1` when when the given tangent vector is purely real, even if it isn't a `Real`. And this test checks that using an actually `Real` tangent vector gives the same result.